x/stakeibc/keeper/unbonding_records.go

```diff
@@ -365,8 +439,17 @@ func (k Keeper) UnbondFromHostZone(ctx sdk.Context, hostZone types.HostZone) err
 		return errorsmod.Wrapf(types.ErrICAAccountNotFound, "no delegation account found for %s", hostZone.ChainId)
 	}
 
-	// Iterate through every unbonding record and sum the total amount to unbond for the given host zone
-	totalUnbondAmount, epochUnbondingRecordIds := k.GetTotalUnbondAmountAndRecordsIds(ctx, hostZone.ChainId)
+	// Get the list of relevant records that should unbond
+	epochUnbondingRecordIds, epochNumberToHostZoneUnbondingMap := k.GetQueuedHostZoneUnbondingRecords(ctx, hostZone.ChainId)
+
+	// Update the native unbond amount on all relevant records
+	// The native amount is calculated from the stTokens
+	if err := k.RefreshUnbondingNativeTokenAmounts(ctx, epochNumberToHostZoneUnbondingMap); err != nil {
+		return err
+	}
+
+	// Sum the total native unbond amount across all records
+	totalUnbondAmount := k.GetTotalUnbondAmount(ctx, epochNumberToHostZoneUnbondingMap)
```

Comment:
Afaiu `GetTotalUnbondAmount` returns the total amount to be unbonded. If so, should we slightly rename and change the comment above the fn def for clarity, maybe `GetTotalAmountToBeUnbonded`?

Afaiu `GetTotalUnbondAmount` only works if passed a a _filtered_ map of `HostZoneUnbonding`s (here, the filtering is performed by `GetQueuedHostZoneUnbondingRecords`. Should we make this filtering requirement more clear?

Reply:
good call, just updated